dask/array/_array_expr/_blockwise.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+
+import numbers
+from collections.abc import Iterable
+
+import numpy as np
+import tlz as toolz
+
+from dask import is_dask_collection
+from dask.array._array_expr._expr import ArrayExpr
+from dask.array._array_expr._utils import compute_meta
+from dask.array.core import (
+    _elemwise_handle_where,
+    _enforce_dtype,
+    apply_infer_dtype,
+    broadcast_shapes,
+    is_scalar_for_elemwise,
+    normalize_arg,
+)
+from dask.blockwise import _blockwise_unpack_collections_task_spec
+from dask.blockwise import blockwise as core_blockwise
+from dask.layers import ArrayBlockwiseDep
+from dask.tokenize import _tokenize_deterministic
+from dask.utils import cached_property, funcname
+
+
+class Blockwise(ArrayExpr):
+    _parameters = [
+        "func",
+        "out_ind",
+        "name",
+        "token",
+        "dtype",
+        "adjust_chunks",
+        "new_axes",
+        "align_arrays",
+        "concatenate",
+        "_meta_provided",
+        "kwargs",
+    ]
+    _defaults = {
+        "name": None,
+        "token": None,
+        "dtype": None,
+        "adjust_chunks": None,
+        "new_axes": None,
+        "align_arrays": True,
+        "concatenate": None,
+        "_meta_provided": None,
+        "kwargs": None,
+    }
+
+    @cached_property
+    def args(self):
+        return self.operands[len(self._parameters) :]
+
+    @cached_property
+    def _meta_provided(self):
+        # We catch recursion errors if key starts with _meta, so define
+        # explicitly here
+        return self.operand("_meta_provided")
+
+    @cached_property
+    def _meta(self):
+        if self._meta_provided is not None:
+            return self._meta_provided
+        else:
+            return compute_meta(self.func, self.dtype, *self.args[::2], **self.kwargs)
+
+    @cached_property
+    def chunks(self):
```

Comment:
Nit, it might be worth a check or todo or assert for unused/unchecked keywords like `align_arrays`.  I imagine the intention is to implement this soon and so we're not bothering for now, but it's probably good practice.  Sometimes we get sidetracked and forget about things when we come back.

(or maybe I'm missing something)

Reply:
No, it's not supported yet. That said, it's the next thing I am working on but it requires so much stuff that I didn't want to add it here to keep the PR somewhat manageable

Reply:
Just a general suggestion / nit of course, not a blocker